### PR TITLE
Refactor Dialogs to inflate views independently

### PR DESCRIPTION
**Pull request by DavDo**
MOB 2814

**Jira issue:**
https://glia.atlassian.net/browse/MOB-2814

**What was solved?**

- Created a mechanism to create GlobalColors from UiTheme. This helped to create AlertTheme from UiTheme and then merge it with provided by UnifiedUi, so now all the Dialog Views are configured with AlertTheme
- Created a mechanism to inflate all the Dialog Views independently and before Dialog is created. This will make all the Views(fully configured with themes) available outside of Alertdialogs and Dialog builders so that these views can be easily tested in snapshot tests.

All these changes will make it easy to change any of these views independently, without affecting other dialogs.

2nd part of improvements are here - https://github.com/salemove/android-sdk-widgets/pull/834

**Release notes:**

 - [ ] Feature
 - [x] Ignore
 - [ ] Release notes (Is it clear from the description here?)
 - [ ] Migration guide (If changes are needed for integrator already using the SDK - what needs to be communicated? Add underneath please)

**Additional info:**

 - [ ] Tests updated? Added? Unit, acceptance, snapshots?
 - [ ] Logging for future troubleshooting of client issues added?

**Screenshots:**

### CODE REVIEW

**Review by andrews-moc — APPROVED** [review]

**Review by AndriiHorishniiMOC — APPROVED** [review]
LGTM!

File: widgetssdk/src/main/java/com/glia/widgets/call/CallView.kt [hunk 1]
```diff
@@ -366,13 +367,12 @@ internal class CallView(
     }
 
     private fun showExitQueueDialog() {
-        alertDialog = Dialogs.showOptionsDialog(
-            context = this.context,
-            theme = theme,
+        val payload = DialogPayload.Option(
```
Comment (Pelkar) [inline widgetssdk/src/main/java/com/glia/widgets/call/CallView.kt:370]: I like this approach

File: widgetssdk/src/main/java/com/glia/widgets/call/CallView.kt [hunk 2]
```diff
@@ -382,49 +382,54 @@ internal class CallView(
...
                 callController?.endEngagementDialogDismissed()
-            },
-            isButtonsColorsReversed = true
-        )
+            }
+        }
     }
 
     private fun showAllowScreenSharingNotificationsAndStartSharingDialog() {
         if (alertDialog == null || !alertDialog!!.isShowing) {
```
Comment (Pelkar) [inline widgetssdk/src/main/java/com/glia/widgets/call/CallView.kt:398]: There is still a bit of duplicated code here. Could you maybe extract the if statement into something more general and instead of showAllowScreenSharingNotificationsAndStartSharingDialog we would have something like createAllowSreenSharingAndStartSharingOptions(): DialogPayload.Option that would create only the option?

File: widgetssdk/src/main/java/com/glia/widgets/call/CallView.kt [hunk 3]
```diff
@@ -707,40 +719,26 @@ internal class CallView(
         }
     }
 
-    private fun showOptionsDialog(
```
Comment (Pelkar) [inline widgetssdk/src/main/java/com/glia/widgets/call/CallView.kt:710]: Why was this deleted, Was this unused or was this merged with some other dialog?

File: widgetssdk/src/main/java/com/glia/widgets/call/CallView.kt [hunk 4]
```diff
@@ -809,12 +807,13 @@ internal class CallView(
     }
 
     private fun showOverlayPermissionsDialog() {
```
Comment (Pelkar) [inline widgetssdk/src/main/java/com/glia/widgets/call/CallView.kt:809]: And shouldn't these dialog type definitions be somewhere else entirely? Why are they defined in View classes. This way they are not reusable. Maybe define them in Dialogs.kt instead and the view only adds listeners?

**Review by Pelkar — CHANGES_REQUESTED** [review]
The overall approach is good, but I would like to have the dialog configurations out of views. That was one of the main problems with dialogs in the first place that we have view specific dialogs

File: widgetssdk/src/main/java/com/glia/widgets/call/CallView.kt (hunk 3 above)
Comment (DavDo) [reply widgetssdk/src/main/java/com/glia/widgets/call/CallView.kt:710]: This was used only once, so I decided to use function from `Dialogs` instead

**Comment by DavDo** [comment]
> The overall approach is good, but I would like to have the dialog configurations out of views. That was one of the main problems with dialogs in the first place that we have view specific dialogs

I agree with all of the suggestions provided, and I will proceed with implementing them. Once I have made the necessary changes, I will request a review again.

**Review by gugalo — APPROVED** [review]

**Comment by DavDo** [comment]
Guys I made most of the requested changes, and separate functions for all the dialog types, so I'm re-requesting your review

**Comment by DavDo** [comment]
2nd part of improvements are here - https://github.com/salemove/android-sdk-widgets/pull/834

**Review by AndriiHorishniiMOC — APPROVED** [review]

**Review by andrews-moc — APPROVED** [review]
Looks good to me. Even better now.

**Review by gugalo — APPROVED** [review]
